WeatherMonitor/ConfigData/configurationData.cs

```diff
@@ -0,0 +1,16 @@
+﻿using WeatherMonitor.BotFactory;
+using WeatherMonitor.Bots;
+namespace WeatherMonitor.ConfigData;
+
+public sealed class ConfigurationData
+{
+    public Bot RainBot { get; set; } = new RainBotFactory().GetBot();
+    public Bot SunBot { get; set; } = new SunBotFactory().GetBot();
+    public Bot SnowBot { get; set; } = new SnowBotFactory().GetBot();
+
+    public static ConfigurationData ConfigurationDataInstance => LazyConfigurationData.Value;
+
+    private static readonly Lazy<ConfigurationData> LazyConfigurationData = new Lazy<ConfigurationData>(new ConfigurationData());
```

Comment:
why you are using lazy here?

Reply:
I am using ```Lazy<T>``` to handle creating singleton object with lazy initialization and thread safety without any additional logic make it more clean and efficient.